clients/web/src/lib/components/atoms/Drawer/Drawer.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+    import {XCircle} from "@steeze-ui/heroicons";
+    import {Icon} from "@steeze-ui/svelte-icon";
+    import {fly, fade, type FlyParams} from "svelte/transition";
+    // Specific import to prevent circular dependency issues
+    import {Portal} from "$lib/components/abstract";
+    import {setContext} from "svelte";
+    import {DRAWER_CLOSE_CONTEXT, type DrawerPosition} from "./types";
+
+    export let open: boolean = false;
+    export let position: DrawerPosition = "left";
+
+    let flyProps: FlyParams = {};
+    $: switch (position) {
+        case "left":
+        default:
+            flyProps = {x: -100};
+            break;
+        case "right":
+            flyProps = {x: 100};
+            break;
+        case "top":
+            flyProps = {y: -100};
+            break;
+        case "bottom":
+            flyProps = {y: 100};
+            break;
+    }
+
+    function handleKeyPress(e: KeyboardEvent) {
+        if (!open) return;
+        if (e.key !== "Escape") return;
+        open = false;
```

Comment:
Stylistic thing, but I think this would be clearer written with just the positive case
We should also use the `close()` function in case some business logic is added there later on

I.e.
```suggestion
        if (open && e.key === "Escape") {
            close()
        }
```